desktop virtualization

datr93 opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

datr93 committed:
desktop virtualization

AutorestCI commented:
/azp run automation - sdk

datr93 commented:
private repo pr https://github.com/Azure/azure-rest-api-specs-pr/pull/1194

azuresdkci commented:
Can one of the admins verify this patch?

anthony-c-martin commented:
> private repo pr [Azure/azure-rest-api-specs-pr#1194](https://github.com/Azure/azure-rest-api-specs-pr/pull/1194)

Signing off for ARM as the private PR has already had sign off.

datr93 commented:
@xccc-msft can you help review the errors?

datr93 commented:
@Azure/azure-sdk-team can someone help review this. We need this merged asap.

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
fix spell check

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
add reseting

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
prettifier

AutorestCI commented:
/azp run automation - sdk

datr93 commented:
@xccc-msft  can you approve asap. We need to go public preview and have powershell generation steps to do on top of this.

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
```
More descriptive

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
```
more descriptive

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          }
+        ],
+        "responses": {
+          "200": {
```
Per previous comments, please differentiate the meaning of these response codes in the descriptions.

markcowl replied:
Also, is this operation long-running (asynchronous)?

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        "tags": [
+          "UserSession"
+        ],
+        "operationId": "UserSessions_List",
```
```suggestion
        "operationId": "UserSessions_ListBySessionHost",

```

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
```
User session successfully disconnected.

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
```
Message successfully sent to user session.

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+    }
+  },
+  "definitions": {
+    "ErrorResponse": {
```
Please use the standard error response, for example: https://github.com/Azure/azure-rest-api-specs/blob/7e9c8a45c28ada7d4a79b14134cb61e6afbcac95/specification/resources/resource-manager/Microsoft.Resources/stable/2019-07-01/resources.json#L4205

markcowl replied:
Generally, errors shoudl also be called 'CloudError'

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        }
+      }
+    },
+    "Resource": {
```
Is there a reason you are not using the common resource models? https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L9

datr93 replied:
it doesn't work

datr93 replied:
other teams are doing this too so i figure it just didn't work.

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          "type": "array",
+          "description": "List of applicationGroup links.",
+          "items": {
+            "type": "string"
```
Are these full resource Ids, or names? Should indicate in the description

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        },
+        "vmTemplate": {
+          "type": "string",
+          "description": "VM template for sessionhosts within hostpool."
```
Is this a resource reference, or seomething else?  Description should indicate.

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        "applicationGroupReferences": {
+          "readOnly": true,
+          "type": "array",
+          "description": "List of applicationGroup links.",
```
resource ids?

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        },
+        "token": {
+          "readOnly": true,
+          "type": "string",
```
Is this meant to be consumed as a string, or as a byte array?  If string this is fine, otherwise, you can add format byte, which indicated this is a base64 encoded string that should be treated as a byte array.

datr93 replied:
ok it is

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        "resetToken": {
+          "readOnly": true,
+          "type": "boolean",
+          "description": "Update registation token."
```
```suggestion
          "description": "Update registration token."

```

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        },
+        "commandLineSetting": {
+          "type": "string",
+          "description": "Specifies whether this published application can be launched with command line arguments provided by the client, command line arguments specified at publish time, or no command line arguments at all."
```
should this be an enum?

datr93 replied:
yes

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        },
+        "commandLineSetting": {
+          "type": "string",
+          "description": "Specifies whether this published application can be launched with command line arguments provided by the client, command line arguments specified at publish time, or no command line arguments at all."
```
enum?

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          "description": "Status for a SessionHost."
+        },
+        "statusTimestamp": {
+          "format": "date-time",
```
read only?

markcowl commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        },
+        "lastUpdateTime": {
+          "format": "date-time",
+          "type": "string",
```
read only?

datr93 committed:
spell check

markcowl commented on custom-words.txt:
```diff
@@ -1749,3 +1749,9 @@ cloudManifestFiles
 kubeadmin
+userprincipalname
+sessionstate
+registation
```
don't need this - should be registration

datr93 replied:
ok

AutorestCI commented:
/azp run automation - sdk

markcowl commented on custom-words.txt:
```diff
@@ -1749,3 +1749,9 @@ cloudManifestFiles
+registation
+sessionhosts
+hostpool
+reseting
```
not necessary, correct spelling is 'resetting'

datr93 replied:
ok

markcowl reviewed (changes requested):
Note:  In general, in responses, please don't use 'OK' as a description for 200, describe this as a successful operation (for example successfully created host pool).  Additionally, there are several places where there are multiple 2xx response status codes for an operation.  This is often an indication that the operation is long-running (asynchronous), so please check theses.  SDKs will nto work as expected if long-running operations are not properly marked.  Also in these cases, please indicate the difference between the success status codes.  I have marked some, but not all of these.

AutorestCI commented:
/azp run automation - sdk

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          "Workspace"
+        ],
+        "operationId": "Workspaces_Get",
+        "description": "Get an workspace.",
```
ok

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "name": "workspaceIn",
```
our api had it as workspaceIn.

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          "Workspace"
+        ],
+        "operationId": "Workspaces_Delete",
+        "description": "Remove an workspace.",
```
ok

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          "Workspace"
+        ],
+        "operationId": "Workspaces_Update",
+        "description": "Update an workspace.",
```
ok

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
```
201 mean created. 200 mean ok

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+              "$ref": "#/definitions/HostPool"
+            }
+          },
+          "201": {
```
201 means created

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+          }
+        ],
+        "responses": {
+          "200": {
```
ok

datr93 commented on specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:
```diff
@@ -0,0 +1,2774 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
```
ok

datr93 committed:
2019-12-10 pr changes
more fixes
add longer description
more 204 descriptions
missed one
need to fix the file renames
example changes
revert all 01 23
one more
revert enum chagne
required and readonly
require the above property
remove require = true
correct example
prettifier
fix properties
more example fixing
more fixes
more issues
remove dup

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
allow

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
fix 01-23 and 09-24

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

allenjzhang commented:
>         "operationId": "Workspaces_Get",

Should be Verb_Noun format.

---
Refers to: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:52 in 213f6ad. [](commit_id = 213f6adcf57272ca2920f562db0d501c61db573a, deletion_comment = False)

allenjzhang commented:
>           "204": {

nit: If no workspace found, do you throw 404 or use 204 to convey that? "No content" description didn't seem very clear.

---
Refers to: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:172 in 213f6ad. [](commit_id = 213f6adcf57272ca2920f562db0d501c61db573a, deletion_comment = False)

allenjzhang commented:
>         "operationId": "Workspaces_ListByResourceGroup",

List_WorkspacesByResourceGroup or just List_Workspaces if you don't have other GET scope such as subscription level.

---
Refers to: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:238 in 213f6ad. [](commit_id = 213f6adcf57272ca2920f562db0d501c61db573a, deletion_comment = False)

datr93 committed:
fix descriptions

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
require x-ms-client-tenant-id

AutorestCI commented:
/azp run automation - sdk

allenjzhang commented:
>   ]

nit: Can you move this to the top of the swagger?

---
Refers to: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:3111 in 213f6ad. [](commit_id = 213f6adcf57272ca2920f562db0d501c61db573a, deletion_comment = False)

datr93 committed:
move stuff to top and add examples for  tenant header

AutorestCI commented:
/azp run automation - sdk

datr93 committed:
remove header

AutorestCI commented:
/azp run automation - sdk
